JavaScript/Angular.md

```diff
@@ -0,0 +1,687 @@
+
+Try to split up your application into components which you can reuse. If, for example, you have to handle lazy loading of all images in the app, create a `Image` module and component in `app/components` and use it in other components whenever you have to show an image. Export the declared component and import/provide dependencies which your component has.
+
+Container components which are loaded directly via routing (by either `component` or `loadChildren`) should be placed in `app/pages` dir. If component is loaded because you have used it in a template, that component should be in `app/components`
+
+One little exception here: if there is some very specific component tied to some routing submodule which is used only in that submodule, it is OK to leave it in that submodule (but as soon as it is used in another routing submodule, move it to `app/components`.
+
+Store all your animations in `app/animations/` and import them for usage in component decorators.
+
+Classes which might be used throughout the application (e.g. generic `Dictionary` class) should be in `app/classes`.
+
+Entity models (`User`, `Post`, etc.) should be in `app/models`.
+
+If you are using some JS libs which do not have typings, add your own custom typing in `app/custom-typings`.
+
+Whenever you have some attribute whose value can be some value from a set of values, make an enum and store that enum in `app/enums`. Good example is an enum for HTTP status codes. If you need some metadata for the enum, create an enum data file which exportst what is basically a dictionary with enum values used as keys and use whatever you need for values. One such example would be: `export const httpStatusCodeData = { [HttpStatus.OK_200]: { translation: 'Success' } }`
+
+If you are using `ngx-form-object`, place `FormObject`s, `FromStore`s and validators in `app/forms`. If you have form components which are reused throughout the app, put them in `app/components/forms`.
+
+Put all route guards in `app/guards`.
+
+If you need some helper functions (like a rounding function which actually rounds correctly *cough* Math.round *cough* floating point *cough*), put the helper in `app/helpers` in its own dir, make sure to write tests for it and simply export it in `app/helpers/helpers.ts`.
+
+Save all your interface files to `app/interfaces`.
+
+If you have some custom pipes for use in templates, create their own dirs in `app/pipes`, similar to helpers. Declare, export and provide the pipes in `app/pipes/pipes.module.ts`.
+
+Services should be placed in `app/services`, each service in its own dir.
+
+If you have any `NgModule`s which are not components (e.g. a module which imports all specific material modules that you need), put them in `app/shared-modules`.
+
+TypeScript `type` declarations should be placed in `app/types`.
```

Comment:
Maybe an example what should be an interface and what should be a type?

Reply:
Added some stuff about types, please check if you agree: https://github.com/infinum/frontend-handbook/blob/310e51e3379ee99a302da3ffdc6e5abf2d9c3b19/JavaScript/Angular.md#prefer-interface-over-type